src/config.ts

```diff
@@ -1,47 +1,54 @@
+import {Collection} from "discord.js";
 import {readFileSync} from "fs";
 import yaml from "js-yaml";
 import {logger} from "./logger";
 import {BotModes} from "./types";
 
+type GuildType = {
+  name: string;
+  features: Array<string>;
+  role_perms: Collection<string, Collection<string, boolean>>;
+};
+
 type ConfigType = {
   api_token: string;
   api_client_id: string;
   bot_user_id: number;
   api_version: string;
   mode: BotModes;
-  development_guild_id: string;
-  debug: boolean;
-  self_roles_channel: string;
-  urls: {mc_address_url: string};
-  prompt: {
-    channel: string;
-    top_text: string;
-    bottom_text: string;
-  };
-  features: {
```

Comment:
im not sure removing this is the best way forward. Having it like as a type that you can call means you wouldnt have issues if you mispell something. 

As an explicit type, I can only use `Config.features.purge` and not mispell it as `Config.features.pruge` because that wouldnt compile.

Reply:
I am fine with the more dynamic nature of the system. In the future, it would fail early because it would try to register a command that doesnt exist.